nep-9.mediawiki

```diff
@@ -0,0 +1,189 @@
+
+Begin transaction to specified address of 1 NEO
+
+<code>neo:AeNkbJdiMx49kBStQdDih7BzfDwyTNVRfb?assetID=c56f33fc6ecfcd0c225c4ab356fee59390af8560be0e930faebe74a6daff7c9b&amount=1.0</code>
+
+Begin transaction to specified address of 1 NEO with transaction attribute description = "Hello"
+
+<code>neo:AeNkbJdiMx49kBStQdDih7BzfDwyTNVRfb?assetID= c56f33fc6ecfcd0c225c4ab356fee59390af8560be0e930faebe74a6daff7c9b&amount=1.0&description=Hello</code>
+
+Begin transaction to specified address of 0.1 GAS. Put public key in ecdh02 attribute field to allow sender to encrypt using ECDH. Transaction attribute description = "Hello"
+
+<code>neo:AQc5mtFayAdoCK13BW1cGAzAHyo9SoUWe7?assetID= 602c79718b16e442de58778e148d0b1084e3b2dffd5de6b7b16cee7969282de7&amount=0.1&ecdh02=02ed53ad58c838435d4dd7a4b25c1eba01384c814ca53a539405434807afbb04b4&description=Hello</code>
+
+
+This should be sufficient to facilitate transfer of native NEO assets safely.
+---
+
+===Smart Contract Invocation for NEP-5 token transfer URI===
+To invoke the smart contract, we need to specify a script hash and the operation of the smart contract that is being invoked as well as providing required parameters. This way an application can tailor the screen to match the invoking operation with pre-filled information. e.g. transfer NEP-5 token, voting, etc. what left for a user to do is to authorize/sign the transaction. It would have the following URI. 
+
+<code>neo:<ScriptHash>?operation=<operation>[&<parameterName>=<parameterValue>][&<TransactionAttributeKey>=<value>]</code>
+
+===Supported Smart Contract URI's===
+
+{| class="wikitable"
+! Number
+! Name
+! Status
+|-
+| 1
+| NEP-5 Token Transfer 
+| Accepted 
+|}
+
+==Rationale==
+As mentioned before, generic smart contract execution is not necessarily suitable for URI's. This is because both of the security issues, and usability. A client of the URI should have 100% confidence that the URI that they will consume will have it's intended effect. Limiting it to a subset of contract operations does most of the heavy lifting. If the client wishes to be even more secure, it can whitelist it's own set of contracts for each subset of supported smart contract URI's.
+
+Any addition to supported smart contract URI's should follow the exact same procedure as other NEO enhancement proposals. In order to be added as a supported smart contract URI, the operation/proposal should have the following... 
+
+1. Not application specific
+2. Well tested
+3. Well documented and easy to consume by a client
+4. A set of examples where the URI is useful
+
+If it meets these criteria than can be considered to enter the list of supported URI schema.
+
+===NEP-5 Token Transfer===
+
+A NEP-5 token transfer is a smart contract invocation and would have the following URI. 
+
+<code>neo:<ScriptHash>?operation=transfer&from=<address>&to=<address>&amount=<amount>[?<TransactionAttributeKey>=<value>]</code>
```

Comment:
It doesn't make sense for a URI to contain the `from` field as anyone clicking on it will want to fill in their own address.

Can we specific if these are optional (`from` and `amount` should be optional)

Reply:
Another note, what endian should the scripthash be in? (I am asking if this should be specified)